stdlib/public/core/Integers.swift.gyb

```diff
@@ -1779,6 +1882,164 @@ ${operatorComment(x.nonMaskingOperator, False)}
   }
 % end
 
+  @_transparent
+  func quotientAndRemainder(
+    dividingBy divisor: Self
+  ) -> (quotient: Self, remainder: Self) {
```

Comment:
Could we use default parameter values for these instead of duplicating the methods?

Reply:
Protocol methods can't have default arguments.

Reply:
The requirement can't, but we can use them in the default implementation.

Reply:
Ah, true!

Reply:
I wonder, though, if that changes overload resolution when operating on a generic. Will it not go through the requirement anymore?

Reply:
Easy enough to test, though we need to be a bit careful here about the distinction between "designed and documented behavior" and "emergent behavior of the constraint solver" =)

Reply:
I think it's "as designed" today (i.e. not just emergent behavior, but a sensible decision given the overloads available), but we could decide to change that design in some way.